src/inter-aut.cc

```diff
@@ -0,0 +1,410 @@
+//
+// Created by Martin Hruska on 10.08.2022.
+//
+
+#include <mata/inter-aut.hh>
+#include <mata/util.hh>
+
+namespace
+{
+    bool is_logical_operator(char ch)
+    {
+        return (Mata::util::haskey(std::set<char>{'&', '|', '!'}, ch));
+    }
+
+    Mata::InterAutomaton::Naming get_naming_type(const std::string &key)
+    {
+        assert(key.find('-') != std::string::npos);
+        const std::string& type = key.substr(key.find('-')+1, std::string::npos);
+
+        if (type == "auto")
+            return Mata::InterAutomaton::Naming::AUTO;
+        else if (type == "enum")
+            return Mata::InterAutomaton::Naming::ENUM;
+        else if (type == "marked")
+            return Mata::InterAutomaton::Naming::MARKER;
+
+        assert(false);
```

Comment:
I'd say it's better to add some string explaining what was wrong, e.g. `assert(false && "Unknown type")`

Reply:
agree